src/libraries/Common/tests/System/IO/ConnectedStreams.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO
+{
+    /// <summary>Provides support for in-memory producer/consumer streams.</summary>
+    internal sealed class ConnectedStreams
+    {
+        /// <summary>Creates a pair of streams that are connected for bidirectional communication.</summary>
+        /// <remarks>Writing to one stream produces data readable by the either, and vice versa.</remarks>
+        public static (Stream Stream1, Stream Stream2) CreateBidirectional() =>
```

Comment:
Any reason you didn't use argument defaults here?

Reply:
From an API design perspective, I don't like baking them in like that.  It limits the ability to change them in the future because any call sites get the values baked in as part of the compiled call site.  For some things, like bools, it makes sense, but for integers on buffer sizes where we could reasonably decide they should change in the future, I think it's better to use overloads.

Reply:
OK